### PR TITLE
Update all dependencies

### PR DESCRIPTION
**Pull request by KTamas**
This updates all dependencies to their latest version, except https://github.com/rt2zz/redux-persist which inexplicably breaks when going from 5.10.0 to 6.0.0. A cursory look at its GH issues shows I'm not the only one having problems with the new major version.

I fixed a few react warnings (missing `readOnly` attribute) and the new versions of `timeago` required some minor code changes.

### CODE REVIEW

**Comment by KTamas** [comment]
I pulled in your latest commits and merged them.

**Comment by KTamas** [comment]
I also fixed a bug in the tests. `new Date(2017, 3, 10)` gives inconsistent results on different computers because timezones.

File: app/js/timeagoLocale.js [hunk 1]
```diff
@@ -1,45 +1,46 @@
 // @flow
+import * as langs from 'timeago.js/lib/lang';
 
 const timeagoLocale = {
-  get ar() { return require('timeago.js/locales/ar'); },
```
Comment (ssorallen) [inline app/js/timeagoLocale.js]: This was doing dynamic requires so the user only has to load the locale they're using. By doing a static `import` at the top, every user will load every locale. Did you intend to change that?

File: package.json [hunk 2]
```diff
@@ -32,35 +32,35 @@
     "babel-eslint": "^10.0.2",
     "babel-jest": "^24.8.0",
     "babel-loader": "^8.0.2",
-    "copy-webpack-plugin": "^4.5.1",
-    "core-js": "^3.1.4",
-    "crowdin-api": "^2.0.2",
-    "css-loader": "^1.0.0",
+    "copy-webpack-plugin": "^5.0.0",
+    "core-js": "3",
```
Comment (ssorallen) [inline package.json]: This was already pegged at `^3.1.4`, could it be at least that or higher?

File: package.json [hunk 3]
```diff
@@ -70,21 +70,21 @@
...
     "memoize-one": "^5.0.5",
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
     "react-redux": "^7.1.0",
-    "react-transition-group": "^2.4.0",
+    "react-transition-group": "^4.0.0",
     "react-virtualized": "^9.21.1",
     "redux": "^4.0.0",
-    "redux-persist": "^5.10.0",
+    "redux-persist": "^5.0.0",
```
Comment (ssorallen) [inline package.json]: Why the downgrade here?

File: app/js/ClosedTabRow.js [hunk 4]
```diff
@@ -60,6 +60,7 @@ export default function ClosedTabRow(props: Props) {
           className="checkbox--td"
           onClick={_handleClickCheckbox}
           type="checkbox"
+          readOnly
```
Comment (ssorallen) [inline app/js/ClosedTabRow.js]: Can you test this out? I remember trying to address this issue a while back, but just adding `readOnly` will make the checkboxes uneditable. This needs some more involved refactoring to get rid of the React warning.

File: package.json [hunk 5]
```diff
@@ -32,35 +32,35 @@
...
+    "css-loader": "^3.0.0",
     "eslint": "^6.1.0",
-    "eslint-plugin-flowtype": "^4.2.0",
+    "eslint-plugin-flowtype": "^4.0.0",
     "eslint-plugin-react": "^7.14.3",
     "fancy-log": "^1.3.3",
-    "file-loader": "^2.0.0",
-    "flow-bin": "^0.105.2",
+    "file-loader": "^5.0.0",
+    "flow-bin": "^0.114.0",
```
Comment (ssorallen) [inline package.json]: I never integrated Flow with CI. Could you try running `yarn flow` after this upgrade and make sure the repo still typechecks? There may be other updates required.

File: app/js/timeagoLocale.js (hunk 1 above)
Comment (KTamas) [reply app/js/timeagoLocale.js]: yeah. the new version of timeago.js changed things, and `require` simply doesn't work anymore, this is the cleanest way. yes there is a tiny overhead with it, but in the grand scheme of things it's not much.

File: package.json (hunk 5 above)
Comment (KTamas) [reply package.json]:
```
~/Prog/tabwrangler(updates-try2 ✔) yarn flow
yarn run v1.21.1
$ /Users/ktamas/Prog/tabwrangler/node_modules/.bin/flow
Launching Flow server for /Users/ktamas/Prog/tabwrangler
Spawned flow server (pid=10477)
Logs will go to /private/tmp/flow/zSUserszSktamaszSProgzStabwrangler.log
Monitor logs will go to /private/tmp/flow/zSUserszSktamaszSProgzStabwrangler.monitor_log
No errors!
✨  Done in 5.68s.
```

File: package.json (hunk 3 above)
Comment (KTamas) [reply package.json]: it installs the latest of version 5 anyways, but i changed it back

File: package.json (hunk 2 above)
Comment (KTamas) [reply package.json]: this also installs the latest from v3 but i pegged it at that version now

File: app/js/ClosedTabRow.js (hunk 4 above)
Comment (KTamas) [reply app/js/ClosedTabRow.js]: I have no issues checking or unchecking checkboxes, just tried.

**Comment by KTamas** [comment]
This should be all good, I think.

**Comment by ssorallen** [comment]
@KTamas, the build is failing because the updated version of file-loader is not compatible with Node v8 being used in Travis:

    error file-loader@5.0.2: The engine "node" is incompatible with this module. Expected version ">= 10.13.0". Got "8.16.2"

It's fine to drop v8 from Travis though, it's an old version of Node at this point. Could you remove that version from `.travis.yml`? Then this should be good to go.

**Comment by KTamas** [comment]
all good!

**Comment by KTamas** [comment]
but now I see that a bunch of new package updates are out, lol. all minor, but i could give them a go before merge.

**Comment by ssorallen** [comment]
This looks great, thanks a lot for all of the updates. Merging merging.

**Pull request merged**